core/chains/evm/logpoller/clients.go

```diff
@@ -0,0 +1,182 @@
+package logpoller
+
+import (
+	"context"
+	"math/big"
+
+	avaclient "github.com/ava-labs/avalanche-rosetta/client"
+	"github.com/ava-labs/coreth/interfaces"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+)
+
+type avaClient struct {
+	aec *avaclient.EthClient
+}
+
+func (a *avaClient) HeaderByNumber(ctx context.Context, number *big.Int) (*Header, error) {
+	header, err := a.aec.HeaderByNumber(ctx, number)
+	if err != nil {
+		return nil, err
+	}
+	return &Header{
+		Hash:       header.Hash(),
+		ParentHash: header.ParentHash,
+		Number:     header.Number,
+	}, nil
+}
+
+func (a *avaClient) HeaderByHash(ctx context.Context, hash common.Hash) (*Header, error) {
+	header, err := a.aec.HeaderByHash(ctx, hash)
+	if err != nil {
+		return nil, err
+	}
+	return &Header{
+		Hash:       header.Hash(),
+		ParentHash: header.ParentHash,
+		Number:     header.Number,
+	}, nil
+}
+
+func (a *avaClient) FilterLogs(ctx context.Context, q ethereum.FilterQuery) ([]types.Log, error) {
+	lgs, err := a.aec.FilterLogs(ctx, interfaces.FilterQuery{
+		BlockHash: q.BlockHash,
+		FromBlock: q.FromBlock,
+		ToBlock:   q.ToBlock,
+		Addresses: q.Addresses,
+		Topics:    q.Topics,
+	})
+	if err != nil {
+		return nil, err
+	}
+	// Cast to geth type
+	var elgs []types.Log
+	for _, lg := range lgs {
+		lgs = append(lgs, lg)
+	}
+	return elgs, err
+}
+
+func (a *avaClient) BatchCallContext(ctx context.Context, b []rpc.BatchElem) error {
+	//TODO implement me
+	panic("implement me")
```

Comment:
will be needed for the VRF, not needed to unblock

Reply:
log poller doesn't do any sends

Reply:
Oh I see this is a new, separate client instance... :thinking: